code/game/gamemodes/gang/gang.dm

```diff
@@ -58,13 +58,19 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
 		gangs += G
 
 		//Now assign a boss for the gang
-		var/datum/mind/boss = pick(antag_candidates)
-		antag_candidates -= boss
-		G.bosses += boss
-		boss.gang_datum = G
-		boss.special_role = "[G.name] Gang Boss"
-		boss.restricted_roles = restricted_jobs
-		log_game("[boss.key] has been selected as the Boss for the [G.name] Gang")
+		for(var/n in 1 to 3)
+			var/datum/mind/boss = pick(antag_candidates)
+			antag_candidates -= boss
+			G.bosses += boss
+			boss.gang_datum = G
+			var/title
+			if(n == 1)
+				title = "Boss"
```

Comment:
I'd rewrite this for to be more like this:
```
var/list/available_ranks = list("Boss" = 1, "Lieutenant" = 2)
for(var/I in available_ranks)
    for(var/J in 1 to available_ranks[I])
        assign_rank_to_candidate(I)
```
And move the stuff inside to for to a new assign_rank_to_candidate proc